zephyr/Kconfig

```diff
@@ -49,6 +49,7 @@ config ZEPHYR_DP_SCHEDULER
 	depends on ZEPHYR_SOF_MODULE
 	depends on ACE
 	depends on PIPELINE_2_0
+	imply XTENSA_HIFI_SHARING if XTENSA_HIFI
```

Comment:
why `imply` and not `select`?

Reply:
@lyakh imply leaves it possible to turn it off https://docs.zephyrproject.org/latest/build/kconfig/tips.html#the-imply-statement . I guess this is up to debata, but sharing does have a perf impact and if you don't you hifi in DP modules, then you strictly speaking don't need this.

Reply:
@lyakh but with loadable modules (potentially out of tree), this won't be enough, right?  There was discussion with the Zephyr PR on the option to make the context save on-demand. I.e. take an exception on hifi use and mark a bit for the thread context. This would add some cost to the common case (e.g. typical LL execution with no preemption of threads between LL tasks).

I think this is now a product policy call and basicly with this PR, you have to go out of your way to disable this on SOF. But given this was disabled before, this construct using "imply" at least allows to revert to that older policy in case anyone needs. If we'd use select,  that would lock the policy with no per-target way out.

Reply:
@kv2019i would it be valid to write
```
select XTENSA_HIFI_SHARING if DP and XTENSA_HIFI
imply XTENSA_HIFI_SHARING if XTENSA_HIFI
```
? But well, yes, there's always a way to break things

Reply:
@lyakh Yes, that could be done as well, the second imply would have to be under some other Kconfig option that is picked by all SOF targets.

That would make it impossible to use the current workarounds:
- run the DP tasks on a dedicated core (in use)
- do not use HIFI in DP tasks

This certainly should not be the default, but whether it's allowed at all, that's a good question.